Binary files libtool-2.4.2.orig/libltdl/config/.install-sh.swp and libtool-2.4.2/libltdl/config/.install-sh.swp differ
diff -urN libtool-2.4.2.orig/libltdl/config/install-sh libtool-2.4.2/libltdl/config/install-sh
--- libtool-2.4.2.orig/libltdl/config/install-sh	2012-04-09 17:46:40.263804500 -0700
+++ libtool-2.4.2/libltdl/config/install-sh	2012-04-09 20:24:38.226804500 -0700
@@ -128,6 +128,10 @@
   RMPROG STRIPPROG
 "
 
+setresult () {
+	return $1
+}
+
 while test $# -ne 0; do
   case $1 in
     -c) ;;
@@ -341,9 +345,11 @@
 	    tmpdir=${TMPDIR-/tmp}/ins$RANDOM-$$
 	    trap 'ret=$?; rmdir "$tmpdir/d" "$tmpdir" 2>/dev/null; exit $ret' 0
 
-	    if (umask $mkdir_umask &&
-		exec $mkdirprog $mkdir_mode -p -- "$tmpdir/d") >/dev/null 2>&1
+	    oldumask=`umask`
+	    umask $mkdir_umask || { error=$? ; echo "Invalid umask: \"${mkdir_umask}\"." >&2 ; setresult $error ; }
+	    if test $? = 0 && exec $mkdirprog $mkdir_mode -p -- "$tmpdir/d" >/dev/null 2>&1
 	    then
+	      umask $oldumask || { error=$? ; echo "Couldn't restore umask \"${oldumask}\"." >&2 ; }
 	      if test -z "$dir_arg" || {
 		   # Check for POSIX incompatibilities with -m.
 		   # HP-UX 11.23 and IRIX 6.5 mkdir -m -p sets group- or
@@ -364,6 +370,7 @@
 	      fi
 	      rmdir "$tmpdir/d" "$tmpdir"
 	    else
+	      umask $oldumask || echo "Couldn't restore umask \"${oldumask}\"." >&2
 	      # Remove any dirs left behind by ancient mkdir implementations.
 	      rmdir ./$mkdir_mode ./-p ./-- 2>/dev/null
 	    fi
@@ -372,10 +379,14 @@
     esac
 
     if
-      $posix_mkdir && (
-	umask $mkdir_umask &&
-	$doit_exec $mkdirprog $mkdir_mode -p -- "$dstdir"
-      )
+      $posix_mkdir && {
+        oldumask=`umask`
+	umask $mkdir_umask || { error=$? ; echo "Invalid umask: \"${mkdir_umask}\"." >&2 ; setresult $error ; }
+	result=$?
+	test $result = 0 && { $doit_exec $mkdirprog $mkdir_mode -p -- "$dstdir" ; result=$? ; }
+        umask $oldumask || echo "Couldn't restore umask \"${oldumask}\"." >&2
+	setresult $result
+      }
     then :
     else
 
@@ -410,8 +421,24 @@
 	  prefixes=
 	else
 	  if $posix_mkdir; then
-	    (umask=$mkdir_umask &&
-	     $doit_exec $mkdirprog $mkdir_mode -p -- "$dstdir") && break
+            umaskunset=no
+            test x${umask-umaskunset} = xumaskunset && umaskunset=yes
+            oldumaskvar="$umask"
+	    umask=$mkdir_umask && \
+	      if $doit_exec $mkdirprog $mkdir_mode -p -- "$dstdir" ; then
+                if test $umaskunset = no ; then
+                  umask="$oldumaskvar"
+                else
+                  unset umask
+		fi
+	        break
+              else
+                if test $umaskunset = no ; then
+                  umask="$oldumaskvar"
+                else
+                  unset umask
+		fi
+	      fi
 	    # Don't fail if two instances are running concurrently.
 	    test -d "$prefix" || exit 1
 	  else
@@ -427,9 +454,16 @@
 
       if test -n "$prefixes"; then
 	# Don't fail if two instances are running concurrently.
-	(umask $mkdir_umask &&
-	 eval "\$doit_exec \$mkdirprog $prefixes") ||
-	  test -d "$dstdir" || exit 1
+        oldumask=`umask`
+	umask $mkdir_umask || { error=$? ; echo "Invalid umask: \"${mkdir_umask}\"." >&2 ; setresult $error ; }
+	result=$?
+	if test $result = 0 ; then
+	  eval "\$doit_exec \$mkdirprog $prefixes") || test -d "$dstdir" || exit 1
+          umask $oldumask || echo "Couldn't restore umask \"${oldumask}\"." >&2
+        else
+          umask $oldumask || echo "Couldn't restore umask \"${oldumask}\"." >&2
+          test -d "$dstdir" || exit 1
+        fi
 	obsolete_mkdir_used=true
       fi
     fi
@@ -450,7 +484,14 @@
     trap 'ret=$?; rm -f "$dsttmp" "$rmtmp" && exit $ret' 0
 
     # Copy the file name to the temp name.
-    (umask $cp_umask && $doit_exec $cpprog "$src" "$dsttmp") &&
+    {
+      oldumask=`umask`
+      umask $cp_umask || { error=$? ; echo "Invalid umask: \"${cp_umask}\"." >&2 ; setresult $error ; }
+      result=$?
+      test $result = 0 && { $doit_exec $cpprog "$src" "$dsttmp" ; result=$? ; }
+      umask $oldumask || echo "Couldn't restore umask \"${oldumask}\"." >&2
+      setresult $result
+    } &&
 
     # and set any options; do chmod last to preserve setuid bits.
     #
@@ -498,7 +539,7 @@
 	    { $doit $rmcmd -f "$rmtmp" 2>/dev/null; :; }
 	  } ||
 	  { echo "$0: cannot unlink or rename $dst" >&2
-	    (exit 1); exit 1
+	    setresult 1; exit 1
 	  }
 	} &&
 
